packages/flutter/lib/src/material/pickers/calendar_date_range_picker.dart

```diff
@@ -0,0 +1,743 @@
+import 'dart:math' as math;
+
+import 'package:flutter/gestures.dart' show DragStartBehavior;
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import 'date_utils.dart' as utils;
+
+const double _monthItemHeaderHeight = 58.0;
+const double _monthItemFooterHeight = 12.0;
+const double _monthItemRowHeight = 42.0;
+const double _monthItemSpaceBetweenRows = 8.0;
+const double _horizontalPadding = 8.0;
+const double _maxCalendarWidthLandscape = 384.0;
+const double _maxCalendarWidthPortrait = 480.0;
+
+/// Displays a scrollable calendar grid that allows a user to select a range
+/// of dates.
+/// 
+/// Note: this is not publicly exported (see pickers.dart), as it is just an
+/// internal component used by [showDateRangePicker].
+class CalendarDateRangePicker extends StatefulWidget {
```

Comment:
It was mostly because I wasn't yet ready to commit to it being part of the public API, but didn't want to add it's bulk to the already long `date_range_picker_dialog.dart`. I really wish dart had a 'package private' access control for things like this.

Reply:
Well, it does have "library private", but the library can only be extended to multiple files with the `part of` directive, which we discourage. I know what you mean though: it would be nice if things could be made private to anything in the pub package.

Like it or not, it's part of the public API, it just isn't exported with the top level include points.  Anyone can use this class with:
```dart
import 'package:/flutter/src/material/pickers/calendar_date_range_picker.dart';
```

Sure, it's not likely, but they could.

Anyhow, it's not a big deal: I suspect that a) people are on their own if they do that anyhow, and 2) that we might want to make this a public API anyhow.

If you're not ready to make it public, though, it's probably OK to put it in the other huge file and make it private. We have plenty of those in Flutter (mostly for that reason) already. `widgets/basic.dart` is 7144 lines...

Reply:
Yeah, I mostly landed on "a) people are on their own if they do that". It is called out as a "DON'T" in the effective dart guide (https://dart.dev/guides/language/effective-dart/usage#dont-import-libraries-that-are-inside-the-src-directory-of-another-package), but I totally get the point that it is technically still public.

I am leaning toward leaving this way for now. That will make it slightly easier if we do want to make it public at some point.